runners/direct-java/src/main/java/org/apache/beam/runners/direct/StatefulParDoEvaluatorFactory.java

```diff
@@ -128,19 +108,6 @@ public void cleanup() throws Exception {
     final DoFn<KV<K, InputT>, OutputT> doFn = application.getTransform().getDoFn();
     final DoFnSignature signature = DoFnSignatures.getSignature(doFn.getClass());
 
-    // If the DoFn is stateful, schedule state clearing.
-    // It is semantically correct to schedule any number of redundant clear tasks; the
-    // cache is used to limit the number of tasks to avoid performance degradation.
-    if (signature.stateDeclarations().size() > 0) {
-      for (final WindowedValue<?> element : inputBundle.getElements()) {
-        for (final BoundedWindow window : element.getWindows()) {
-          cleanupRegistry.get(
-              AppliedPTransformOutputKeyAndWindow.create(
-                  application, (StructuralKey<K>) inputBundle.getKey(), window));
-        }
-      }
-    }
-
```

Comment:
This is removed because it is somewhat duplicate to what StatefulDoFnRunner already does. We can return it back, it is not directly related to purpose of this PR, it is just a cleanup that I did during bug hunting.

Reply:
Please do make it a separate commit. Because when reading code history it will make more sense. Also single commits can be rolled back. it is fine to have it in the same PR in my opinion.

I agree that this seems redundant with https://github.com/apache/beam/blob/c3d1e5d71ada0097a48d235cb717c61f63b5eb80/runners/direct-java/src/main/java/org/apache/beam/runners/direct/ParDoEvaluator.java#L113

This code I think predates `StatefulDoFnRunner` so perhaps this was missed when it was ported to use `StatefulDoFnRunner`. (I could be wrong; I did not check the git history)

Reply:
:+1: